### PR TITLE
feat(web): dispute-maintenance-buttons

### LINKED ISSUE #1601: UI: add buttons for dispute-level maintenance
The Next Period button calls `KC.passPeriod(dispute.id)`
The Execute button calls `KC.executeRuling(dispute.id)`
The Distribute Rewards/Penalties button calls `KC.execute(dispute.id, dispute.currentRoundIndex, 20)`

If the prepareCall fails, make the corresponding button greyed out (disabled), possibly show the error message somewhere (in the console or in a tooltip?).

image

### PR DESCRIPTION
**Pull request by Harman-singh-waraich**
- adds dispute maintenance buttons to manually do operations in case bots fails or in development.
- The ripple shows when a Period is over or Execution is pending. Not showing it for  rewards distribution, since that is not straightforward to determine and want to keep these buttons minimal.

closes #1601 

## Summary by CodeRabbit

- **New Features**
  - Introduced `MaintenanceButtons` component for managing maintenance actions related to disputes.
  - Added action buttons including `DrawButton`, `ExecuteRulingButton`, and `PassPeriodButton`.
  - Implemented `DistributeRewards` component for reward distribution management.
  - Added `useTransactionBatcher` custom hook to facilitate batching of contract transactions.

- **Enhancements**
  - Updated the `CaseDetails` page structure to incorporate the `MaintenanceButtons`.
  - Added `nbVotes` field to the `disputeDetailsQuery` for improved tracking of disputes.
  - Enhanced GraphQL schema with new fields related to fee management and dispute resolution.

---

## PR-Codex overview
This PR introduces various updates related to dispute details, maintenance, and contract batching. 

### Detailed summary
- Added new fields `jurorsDrawn` and `jurorRewardsDispersed` to rounds
- Implemented `TransactionBatcher` contract for batch transactions
- Updated UI components for maintenance buttons and case details

> The following files were skipped due to too many changes: `web/src/pages/Cases/CaseDetails/MaintenanceButtons/MenuButton.tsx`, `web/src/pages/Cases/CaseDetails/MaintenanceButtons/ExecuteRuling.tsx`, `subgraph/core/src/KlerosCore.ts`, `web/src/hooks/useTransactionBatcher.tsx`, `web/src/pages/Cases/CaseDetails/MaintenanceButtons/DrawButton.tsx`, `web/src/pages/Cases/CaseDetails/MaintenanceButtons/PassPeriodButton.tsx`, `web/src/pages/Cases/CaseDetails/MaintenanceButtons/DistributeRewards.tsx`, `web/src/pages/Cases/CaseDetails/MaintenanceButtons/index.tsx`, `web/src/pages/Cases/CaseDetails/MaintenanceButtons/WithdrawAppealFees.tsx`, `contracts/deployments/arbitrum/TransactionBatcher.json`, `contracts/deployments/arbitrumSepoliaDevnet/TransactionBatcher.json`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Review by kemuru — DISMISSED** [review]

**Review by kemuru — APPROVED** [review]

**Review by alcercu — DISMISSED** [review]
lgtm

File: web/src/pages/Cases/CaseDetails/MaintenanceButtons/index.tsx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+      if (dispute.ruled) {
+        clearInterval(intervalId);
+        return;
+      }
+      rippleCheck();
+    }, 5000);
+
+    return () => clearInterval(intervalId);
+  }, [dispute]);
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/pages/Cases/CaseDetails/MaintenanceButtons/index.tsx:96]: rippleCheck is called at top-level , before SetInterval so the check is necessary
in setInterval the check is needed to make sure dispute is loaded first.

**Comment by Harman-singh-waraich** [comment]
- Implementing better button handling
- hook update
Merge on hold

File: web/src/pages/Cases/CaseDetails/MaintenanceButtons/DrawButton.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  const {
+    data: drawConfig,
+    isLoading: isLoadingConfig,
+    isError,
+  } = useSimulateKlerosCoreDraw({
+    query: {
+      enabled: !isUndefined(id) && !isUndefined(numberOfVotes),
+    },
+    args: [BigInt(id ?? 0), BigInt(numberOfVotes ?? 0)],
+  });
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/pages/Cases/CaseDetails/MaintenanceButtons/DrawButton.tsx:36]: handled within hook

File: web/src/pages/Cases/CaseDetails/MaintenanceButtons/WithdrawAppealFees.tsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      for (let round = roundIndex; round >= 0; round--) {
+        argsArr.push({
+          ...baseArgs,
+          args: [BigInt(id), contribution.contributor.id, BigInt(round), contribution.choice],
+        });
+      }
+    }
+
+    setContractConfigs(argsArr);
+  }, [id, roundIndex, chainId, filteredContributions]);
```
[the remark this answers is not in the document]
Comment (Harman-singh-waraich) [reply web/src/pages/Cases/CaseDetails/MaintenanceButtons/WithdrawAppealFees.tsx:79]: required for determining disputeKit address

**Review by jaybuidl — APPROVED** [review]

**Pull request merged**